### PR TITLE
[Tabs] Increase badge text contrast.

### PR DESCRIPTION
**Pull request by romoore**
The Tabs badge default text has a contrast ratio of around 4:1. For WCAG
AA-level text, we should have 4.5:1 for small text like this.  Changing
to Material Red #700 increases contrast to 5:1. Material Red #600 only
provides 4.4:1 against white.

Part of #6981

### CODE REVIEW

File: components/Tabs/src/private/MDCItemBarBadge.m [hunk 1]
```diff
@@ -36,7 +37,7 @@ - (instancetype)initWithFrame:(CGRect)frame {
 
 - (void)commonMDCTabBarItemBadgeInit {
   if (!_badgeColor) {
-    _badgeColor = [UIColor redColor];
+    _badgeColor = MDCPalette.redPalette.tint700;
```
Comment (andrewoverton) [inline components/Tabs/src/private/MDCItemBarBadge.m:40]: I know it's bad to use MDC color scheme defaults as default colors in components. I guess it's okay to do that with MDCPalette colors?

Comment (romoore) [reply components/Tabs/src/private/MDCItemBarBadge.m:40]: Using the Material Palette colors for defaults is very often done in our components. We currently do this in:

* **ActivityIndicator** - Default cycle colors
* **NavigationDrawer** - Default handle color
* **ProgressView** - Default indicator color
* **Slider** - Default track color
* **TextFields** - Default error color

And more.

Comment (andrewoverton) [reply components/Tabs/src/private/MDCItemBarBadge.m:40]: I wasn't sure if it was still considered best practice. SGTM.

**Review by andrewoverton — APPROVED** [review]
Conflicts :/ Approved

File: components/Tabs/src/private/MDCItemBarBadge.m (hunk 1 above)
Comment (romoore) [reply components/Tabs/src/private/MDCItemBarBadge.m:40]: I believe when we have to choose default colors, we prefer Material palette colors.

**Pull request merged**